File: share-code/bai-tap-lap-trinh/python/comprehensive/70.py
your_number = 999

units = ['', 'một', 'hai', 'ba', 'bốn', 'năm', 'sáu', 'bảy', 'tám', 'chín', 'mười']
tens  = ['', 'mười', 'hai mươi', 'ba mươi', 'bốn mươi', 'năm mươi', 'sáu mươi', 'bảy mươi', 'tám mươi', 'chín mươi']

hundreds    = your_number // 100
remainder   = your_number % 100
tens_digit  = remainder // 10
units_digit = remainder % 10

result = ''

if hundreds > 0:
    result += units[hundreds] + ' trăm '

if remainder > 0:
    if remainder < 10:
        result += units[remainder]
    elif remainder < 20:
        result += 'mười ' + units[remainder % 10]
    else:
        result += tens[tens_digit]
        if units_digit > 0:
            result += ' ' + units[units_digit]

print(f"{your_number}: {result}")

File: share-code/bai-tap-lap-trinh/cpp/70.cpp
// Hàm để đọc số nguyên dương
#include <iostream>
using namespace std;
int main() {
    int yourNumber = 42;
    const string one_to_nineteen[] = {"", "một", "hai", "ba", "bốn", "năm", "sáu", "bảy", "tám", "chín", "mười", "mười một", "mười hai", "mười ba", "mười bốn", "mười lăm", "mười sáu", "mười bảy", "mười tám", "mười chín"};
    const string tens[] = {"", "", "hai mươi", "ba mươi", "bốn mươi", "năm mươi", "sáu mươi", "bảy mươi", "tám mươi", "chín mươi"};
    string result = "";

    if (yourNumber >= 1 && yourNumber <= 99) {
        if (yourNumber < 20) {
            result = one_to_nineteen[yourNumber];
        } else {
            result = tens[yourNumber / 10] + " " + one_to_nineteen[yourNumber % 10];
        }
    }
    cout << result << endl;
    return 0;
}